### PR TITLE
[integration] Python 3 compatibility fixes

### PR DESCRIPTION
**Pull request by chrisburr**
Mostly fixes bytes/str mix-ups in the code which are hard to spot without running it. Hopefully this PR systematically finds them all though it's big enough that I'm bound to have made mistakes.

There is a lot of potential for clean up and simplifying the code however I don't want to make this PR any bigger than it already is.

BEGINRELEASENOTES

FIX: Mostly fixes bytes/str mix-ups in the code which are hard to spot without running it; optimization
NEW: add git Python 3 tests

ENDRELEASENOTES

### CODE REVIEW

File: src/WebAppDIRAC/WebApp/handler/UPHandler.py [hunk 1]
```diff
@@ -301,15 +265,8 @@ def web_listPublicStates(self):
...
-    try:
-      name = self.request.arguments['name'][-1]
-    except KeyError as excp:
-      raise WErr(400, "Missing %s" % excp)
-    try:
-      access = self.request.arguments['access'][-1].upper()
-    except KeyError as excp:
-      access = 'ALL'
-
+    name = self.get_argument("name")
```
Comment (TaykYoku) [inline src/WebAppDIRAC/WebApp/handler/UPHandler.py:268]: Why do not raise `WErr` here when argument is absent?

Comment (chrisburr) [reply src/WebAppDIRAC/WebApp/handler/UPHandler.py:268]: `get_argument` raises a `MissingArgumentError` which is a subclass of `HTTPError` so I think there is no need to explicitly raise a `WErr`?

* https://www.tornadoweb.org/en/branch5.1/web.html#tornado.web.RequestHandler.get_argument

File: src/WebAppDIRAC/WebApp/handler/ProxyManagerHandler.py [hunk 2]
```diff
@@ -108,93 +107,65 @@ def web_deleteProxies(self):
...
+    users = list(json.loads(self.get_argument("username", "[]")))
+    if users:
+      req['UserName'] = users
+
+    usersgroup = list(json.loads(self.get_argument("usergroup", "[]")))
+    if usersgroup:
+      req['UserGroup'] = usersgroup
+
+    persistent = self.get_argument("persistent", "")
+    if persistent in ["True", "False"]:
```
Comment (TaykYoku) [inline src/WebAppDIRAC/WebApp/handler/ProxyManagerHandler.py]: Is there a reason to change the logic here?
```suggestion
    if usersgroup and persistent in ["True", "False"]:
```

**Review by TaykYoku — APPROVED** [review]
Looking at it, it looks like a good job

File: src/WebAppDIRAC/WebApp/handler/ProxyManagerHandler.py (hunk 2 above)
Comment (chrisburr) [reply src/WebAppDIRAC/WebApp/handler/ProxyManagerHandler.py]: Well spotted! I hadn't noticed the structure was inconsistent with other similar lines.

I'm not sure if this was a bug as the original code used both `"usergroup"` and `"usersgroup"` and `"usersgroup"` doesn't appear anywhere else in the code:

```python
    if "usergroup" in self.request.arguments:
      usersgroup = list(json.loads(self.request.arguments['usergroup'][-1]))
      if len(usersgroup) > 0:
        req['UserGroup'] = usersgroup

    if "usersgroup" in self.request.arguments and len(self.request.arguments["persistent"]) > 0:
      if str(self.request.arguments["persistent"]) in ["True", "False"]:
        req["PersistentFlag"] = str(self.request.arguments["persistent"])
```

and most lines like this looked like below with the variable name being listed three times:

```python
    if "start" in self.request.arguments and len(self.request.arguments["start"][0]) > 0:
      start = int(self.request.arguments["start"][0])
```

**Pull request merged**